include/library/spdm_transport_tcp_lib.h

```diff
@@ -0,0 +1,169 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2025 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#ifndef SPDM_TCP_TRANSPORT_LIB_H
+#define SPDM_TCP_TRANSPORT_LIB_H
+
+#include "library/spdm_common_lib.h"
+
+#define LIBSPDM_TCP_ALIGNMENT 4
+
+#define LIBSPDM_TCP_SEQUENCE_NUMBER_COUNT 0
+#define LIBSPDM_TCP_MAX_RANDOM_NUMBER_COUNT 0
+
+/* Required sender/receive buffer in device io.
+ * +-------+--------+---------------------------+------+--+------+---+--------+-----+
+ * | TYPE  |TransHdr|      EncryptionHeader     |AppHdr|  |Random|MAC|AlignPad|FINAL|
+ * |       |        |SessionId|SeqNum|Len|AppLen|      |  |      |   |        |     |
+ * +-------+--------+---------------------------+------+  +------+---+--------+-----+
+ * |  TCP  |    4   |    4    |   0  | 2 |   2  |   0  |  |   0  | 16|   3    |  31 |
+ * +-------+--------+---------------------------+------+--+------+---+--------+-----+
+ */
+#define LIBSPDM_TCP_TRANSPORT_ADDITIONAL_SIZE        (12 + \
+                                                      LIBSPDM_PCI_DOE_SEQUENCE_NUMBER_COUNT + \
+                                                      LIBSPDM_PCI_DOE_MAX_RANDOM_NUMBER_COUNT + \
+                                                      LIBSPDM_MAX_AEAD_TAG_SIZE + \
+                                                      (LIBSPDM_PCI_DOE_ALIGNMENT - 1))
+
+#pragma pack(1)
+
+typedef struct {
+    uint16_t payload_length;
+    uint8_t reserved;
+    uint8_t message_type;
+} tcp_spdm_binding_header_t;
+
+#define TCP_MESSAGE_TYPE_OUT_OF_SESSION 0x01
+#define TCP_MESSAGE_TYPE_IN_SESSION 0x02
+#define TCP_MESSAGE_TYPE_HANDSHAKE_REQUEST 0x03
```

Comment:
The definition is incorrect according to the spec.

Reply:
Yes, i have modified them to be as per the spec now.

```
#define TCP_MESSAGE_TYPE_OUT_OF_SESSION 0x05
#define TCP_MESSAGE_TYPE_IN_SESSION 0x06
#define TCP_MESSAGE_TYPE_ROLE_INQUIRY 0xBF

/* Error Messages*/
#define TCP_MESSAGE_TYPE_ERROR_TOO_LARGE 0xC0
#define TCP_MESSGAE_TYPE_ERROR_NOT_SUPPORTED 0xC1
#define TCP_MESSAGE_TYPE_ERROR_CANNOT_OPERATE_AS_REQUESTER 0xC2
#define TCP_MESSAGE_TYPE_ERROR_CANNOT_OPERATE_AS_RESPONDER 0xC3
```

Would it be the right place to put the error codes ? as well ?